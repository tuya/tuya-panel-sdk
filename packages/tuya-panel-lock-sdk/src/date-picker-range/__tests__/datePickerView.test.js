/* eslint-disable jest/expect-expect */
import React from 'react';
import { shallow } from 'enzyme';
import DatePickerRange from '../index';
import DatePicker from '../datePicker';

describe('detePickerView components', () => {
  it('basic render', () => {
    shallow(
      <DatePickerRange
        mode="date"
        startDate={new Date()}
        endDate={new Date()}
        onDateChange={jest.fn()}
      />
    );
    // expect(wrapper).toMatchSnapshot();
  });
  it('default value render', () => {
    const setup = (props = {}) => {
      return shallow(
        <DatePickerRange mode="datetime" startDate="" endDate="" onDateChange={jest.fn()} />
      );
    };
    // expect(wrapper).toMatchSnapshot();
    const wrapper = setup();
    wrapper.instance().openModal(1);
    wrapper.instance().confirmModal();
    wrapper.instance().openModal(0);
    wrapper.instance().confirmModal();
    wrapper.instance().formatDate(new Date());
    wrapper.instance().setDate(new Date());
    wrapper.instance().componentWillReceiveProps({ startDate: new Date(), endDate: new Date() });
    wrapper.instance().openModal(1);
    wrapper.instance().confirmModal();
    wrapper.instance().setDate(new Date('2019-10-11'));
    wrapper.instance().openModal(0);
    wrapper.instance().confirmModal();
  });
  it('TouchableOpacity event', () => {
    const setup = (props = {}) => {
      return shallow(
        <DatePickerRange mode="hour" startDate="" endDate="" onDateChange={jest.fn()} />
      );
    };
    const wrapper = setup();
    wrapper.instance().setDate(new Date());
    wrapper.instance().openModal(0);
    wrapper.instance().confirmModal();
    wrapper.instance().formatDate(new Date());
    wrapper.instance().setDate(new Date());
    wrapper.instance().componentWillReceiveProps({ startDate: new Date(), endDate: new Date() });
    wrapper.instance().openModal(1);
    wrapper.instance().setDate(new Date('2022-10-11'));
    wrapper.instance().confirmModal();
  });
  it('datePicker picker event', () => {
    const setup = (props = {}) => {
      return shallow(
        <DatePicker
          mode="datetime"
          date={new Date()}
          onDateChange={jest.fn()}
          minDate={new Date()}
          maxDate={new Date()}
        />
      );
    };
    const wrapper = setup();
    wrapper.instance().componentDidMount();
    wrapper.instance().componentWillReceiveProps({ date: new Date() });
    wrapper.instance().onValueChange(1, 1, 1);
    wrapper.instance().getNewDate(1, 1, 1);
    wrapper.instance().getDate();
    wrapper.instance().getTimeColsData(new Date());
    wrapper.instance().changeIndexAndCols(new Date(2022 - 11 - 11));
    wrapper.instance().getDateColsData();
    wrapper.instance().getIndexAndCols();
  });
  it('date picker event', () => {
    const setup = (props = {}) => {
      return shallow(
        <DatePicker
          mode="date"
          date={new Date()}
          onDateChange={jest.fn()}
          minDate={new Date(2020 - 11 - 11)}
          maxDate={new Date(2022 - 12 - 12)}
        />
      );
    };
    const wrapper = setup();
    wrapper.instance().componentDidMount();
    wrapper.instance().componentWillReceiveProps({ date: new Date() });
    wrapper.instance().onValueChange(1, 1, 1);
    wrapper.instance().getNewDate(1, 1, 1);
    wrapper.instance().getDate();
    wrapper.instance().getTimeColsData(new Date());
    wrapper.instance().changeIndexAndCols(new Date(2022 - 11 - 11));
    wrapper.instance().getDateColsData();
    wrapper.instance().getIndexAndCols();
  });
  it('date picker event', () => {
    const setup = (props = {}) => {
      return shallow(
        <DatePicker
          mode="hour"
          date={new Date()}
          onDateChange={jest.fn()}
          minDate={new Date()}
          maxDate={new Date()}
        />
      );
    };
    const wrapper = setup();
    wrapper.instance().componentDidMount();
    wrapper.instance().componentWillReceiveProps({ date: new Date() });
    wrapper.instance().onValueChange(1, 1, 1);
    wrapper.instance().getNewDate(1, 1, 1);
    wrapper.instance().getDate();
    wrapper.instance().getTimeColsData(new Date());
    wrapper.instance().changeIndexAndCols(new Date(2022 - 11 - 11));
    wrapper.instance().getDateColsData();
    wrapper.instance().getIndexAndCols();
    wrapper.instance().setHours(new Date(), 1);
  });
});
