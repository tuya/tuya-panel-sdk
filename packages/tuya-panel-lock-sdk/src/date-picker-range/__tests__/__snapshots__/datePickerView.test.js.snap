// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defendTime components basic render 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#333",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        2021-06-29 10:17
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#333",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        2021-06-29 10:17
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:17:37.800Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:17:37.798Z}
      minDate={2021-06-29T02:17:37.800Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Styled(TouchableOpacity)
        onPress={[Function]}
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </Styled(TouchableOpacity)>
      <Styled(TouchableOpacity)
        onPress={[Function]}
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </Styled(TouchableOpacity)>
    </View>
  </Portal>
</View>
`;

exports[`defendTime components cancel TouchableOpacity event 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:33:08.004Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:33:08.004Z}
      minDate={2021-06-29T02:33:08.004Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </TouchableOpacity>
    </View>
  </Portal>
</View>
`;

exports[`defendTime components confirm TouchableOpacity event 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:33:08.013Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:33:08.013Z}
      minDate={2021-06-29T02:33:08.013Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </TouchableOpacity>
    </View>
  </Portal>
</View>
`;

exports[`defendTime components default value render 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:17:37.883Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:17:37.883Z}
      minDate={2021-06-29T02:17:37.883Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Styled(TouchableOpacity)
        onPress={[Function]}
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </Styled(TouchableOpacity)>
      <Styled(TouchableOpacity)
        onPress={[Function]}
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </Styled(TouchableOpacity)>
    </View>
  </Portal>
</View>
`;

exports[`detePickerView components basic render 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#333",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        2021-06-29 10:37
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#333",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        2021-06-29 10:37
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:37:50.887Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:37:50.886Z}
      minDate={2021-06-29T02:37:50.887Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </TouchableOpacity>
    </View>
  </Portal>
</View>
`;

exports[`detePickerView components cancel TouchableOpacity event 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:37:50.993Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:37:50.993Z}
      minDate={2021-06-29T02:37:50.993Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </TouchableOpacity>
    </View>
  </Portal>
</View>
`;

exports[`detePickerView components confirm TouchableOpacity event 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:37:51.000Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:37:51.000Z}
      minDate={2021-06-29T02:37:51.000Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </TouchableOpacity>
    </View>
  </Portal>
</View>
`;

exports[`detePickerView components default value render 1`] = `
<View>
  <View
    key="生效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      生效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <View
    key="失效时间"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 58,
        "justifyContent": "space-between",
        "paddingHorizontal": 5,
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      type={null}
      weight={null}
    >
      失效时间
    </TYText>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Object {
            "color": "#d6d9dc",
            "marginRight": 5,
          }
        }
        type={null}
        weight={null}
      >
        请选择时间
      </TYText>
      <Component
        name="arrow"
      />
    </TouchableOpacity>
  </View>
  <Portal
    onMaskPress={[Function]}
    visible={false}
  >
    <DatePicker
      accessibilityLabel="DatePicker"
      date={2021-06-29T02:37:50.975Z}
      dateSortKeys={null}
      isAmpmFirst={false}
      isPlusZero={true}
      isTimeFirst={false}
      loop={false}
      maxDate={2030-06-29T02:37:50.974Z}
      minDate={2021-06-29T02:37:50.975Z}
      mode="datetime"
      onDateChange={[Function]}
      onValueChange={[Function]}
      pickerFontColor="#333"
      use12Hours={false}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          取消
        </TYText>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "#f8f8f1",
            "borderRightWidth": 0.5,
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          type={null}
          weight={null}
        >
          确定
        </TYText>
      </TouchableOpacity>
    </View>
  </Portal>
</View>
`;
